central/debug/service/service.go

```diff
@@ -540,7 +544,7 @@ func (s *serviceImpl) writeZippedDebugDump(ctx context.Context, w http.ResponseW
 	}
 
 	if s.telemetryGatherer != nil && opts.telemetryMode > 0 {
-		telemetryData := s.telemetryGatherer.Gather(ctx, opts.telemetryMode >= 2)
+		telemetryData := s.telemetryGatherer.Gather(ctx, opts.telemetryMode >= 2, opts.withCentral)
```

Comment:
I cannot comment on there, but you also need to disable ll.566-576 (new) when `opts.withCentral` is false.